lambdas/accesstoken/src/main/java/uk/gov/di/ipv/cri/passport/accesstoken/validation/TokenRequestValidator.java

```diff
@@ -0,0 +1,117 @@
+package uk.gov.di.ipv.cri.passport.accesstoken.validation;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.oauth2.sdk.ParseException;
+import com.nimbusds.oauth2.sdk.auth.JWTAuthenticationClaimsSet;
+import com.nimbusds.oauth2.sdk.auth.PrivateKeyJWT;
+import com.nimbusds.oauth2.sdk.auth.verifier.ClientAuthenticationVerifier;
+import com.nimbusds.oauth2.sdk.auth.verifier.InvalidClientException;
+import com.nimbusds.oauth2.sdk.id.Audience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.di.ipv.cri.passport.accesstoken.domain.ConfigurationServicePublicKeySelector;
+import uk.gov.di.ipv.cri.passport.accesstoken.exceptions.ClientAuthenticationException;
+import uk.gov.di.ipv.cri.passport.library.service.ConfigurationService;
+
+import java.time.OffsetDateTime;
+import java.util.Date;
+import java.util.Map;
+import java.util.Set;
+
+public class TokenRequestValidator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TokenRequestValidator.class);
+    private static final String CLIENT_ASSERTION_PARAM = "client_assertion";
+    private static final String CLIENT_ID_PARAM = "client_id";
+    private static final String NONE = "none";
+    private static final String JWT = "jwt";
+
+    private final ConfigurationService configurationService;
+
+    private final ClientAuthenticationVerifier<Object> verifier;
+
+    private String clientId;
+
+    public TokenRequestValidator(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+        this.verifier = getClientAuthVerifier(configurationService);
+    }
+
+    public void authenticateClient(String requestBody, Map<String, String> queryParams)
+            throws ClientAuthenticationException {
+        if (!queryParams.containsKey(CLIENT_ASSERTION_PARAM)) {
+            if (!queryParams.containsKey(CLIENT_ID_PARAM)) {
+                LOGGER.error(
+                        "Missing either client_assertion or client_id values in request. Failed to establish client_id value.");
+                throw new ClientAuthenticationException(
+                        "Unknown client, no client_id value or client_assertion jwt found in request");
+            } else {
+                clientId = queryParams.get(CLIENT_ID_PARAM);
+
+                String clientAuthenticationMethod = getClientAuthenticationMethod(clientId);
+
+                if (clientAuthenticationMethod.equals(JWT)) {
+                    LOGGER.error("Missing client_assertion jwt for configured client {}", clientId);
+                    throw new ClientAuthenticationException(
+                            String.format(
+                                    "Missing client_assertion jwt for configured client '%s'",
+                                    clientId));
+                }
+            }
+        } else {
```

Comment:
There are quite a few branches in this making it hard to follow - could it be extracted into a well named private method, or broken down in some other way?